#!/usr/bin/env python
PACKAGE = "liquid_height_estimation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_x",    double_t,   0,  "Minimum x-axis field filter value",    0.1,    -5,  5)
gen.add("max_x",    double_t,   0,  "Maximum x-axis field filter value",    0.7,    -5,  5)
gen.add("min_y",    double_t,   0,  "Minimum y-axis field filter value",    -0.4,    -5,  5)
gen.add("max_y",    double_t,   0,  "Maximum y-axis field filter value",    0.4,    -5,  5)
gen.add("min_z",    double_t,   0,  "Minimum z-axis field filter value",    0.7,    -5,  5)
gen.add("max_z",    double_t,   0,  "Maximum z-axis field filter value",    0.9,    -5,  5)

gen.add("min_r",    int_t,    0, "Min Red Value", 0,  0, 255)
gen.add("min_g",    int_t,    0, "Min Green Value", 100,  0, 255)
gen.add("min_b",    int_t,    0, "Min Blue Value", 38,  0, 255)
gen.add("max_r",    int_t,    0, "Max Red Value", 160,  0, 255)
gen.add("max_g",    int_t,    0, "Max Green Value", 200,  0, 255)
gen.add("max_b",    int_t,    0, "Max Blue Value", 131,  0, 255)

exit(gen.generate(PACKAGE, "height_detector", "HeightDetector"))